File: algorithms/easy/longest_common_prefix/longest_common_prefix.php
<?php
class Solution {

    /**
     * @param String[] $strs
     * @return String
     */
    function longestCommonPrefix($strs) {
        
        if(count($strs) == 0) return "";
        if(count($strs) == 1) return $strs[0];
        
        $min = PHP_INT_MAX;
        $min_pos = 0;
        
        for($j = 0; $j < count($strs); $j++){
            if(strlen($strs[$j]) < $min){
                $min = strlen($strs[$j]);
                $min_pos = $j;
            }
        }
        
        $min_strs = str_split($strs[$min_pos]);
        
        $same = true;
        $pos = 0;
        for($k = 0; $k < $min; $k++) {
            for($l = 0; $l < count($strs); $l++){
                $m = str_split($strs[$l]);
                if($m[$k] != $min_strs[$k]) $same = false;
            }
            if(!$same) break;
            $pos++;
        }
        
        $result = "";
        for($n = 0; $n < $pos; $n++){
            $result .= $min_strs[$n];
        }
        
        return $result;
    }
}
?>

File: algorithms/easy/longest_common_prefix/longest_common_prefix.go
package main

import (
	"strings"
)

func longestCommonPrefix(strs []string) string {

	if len(strs) == 0 {
		return ""
	}
	if len(strs) == 1 {
		return strs[0]
	}

	minSize := int(^uint(0) >> 1)
	minPos := 0

	for i := 0; i < len(strs); i++ {
		if len(strs[i]) < minSize {
			minSize = len(strs[i])
			minPos = i
		}
	}

	minStrs := strings.Split(strs[minPos], "")

	same := true
	pos := 0
	for j := 0; j < minSize; j++ {
		for k := 0; k < len(strs); k++ {
			m := strings.Split(strs[k], "")
			if m[j] != minStrs[j] {
				same = false
			}
		}
		if !same {
			break
		}
		pos++
	}

	result := ""
	for l := 0; l < pos; l++ {
		result += minStrs[l]
	}

	return result
}

func main() {
}
